src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0013Tests.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+            await foreach (var x in GetValuesAsync()./*MM*/ConfigureAwait(true)) { }
+        }
+
+        private static async IAsyncEnumerable<int> GetValuesAsync() { yield break; }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0013", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0013NoWarningOnAsyncEnumerableExistingConfigureAwaitTrue()
+        {
+
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+#pragma warning disable AZC0013
+            await foreach (var x in /*MM*/GetValuesAsync().ConfigureAwait(true)) { }
+#pragma warning restore AZC0013
+        }
+
+        private static async IAsyncEnumerable<int> GetValuesAsync() { yield break; }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            Assert.Empty(diagnostics);
+        }
+
+        [Fact]
+        public async Task AZC0013WarningOnAsyncUsingExistingConfigureAwaitTrue()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+            var ad = new AsyncDisposable();
+            await using(ad./*MM*/ConfigureAwait(true)) { }
+        }
+    
+        private class AsyncDisposable : IAsyncDisposable
+        {
+            public ValueTask DisposeAsync() => new ValueTask();
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0013", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0013NoWarningOnAsyncUsingExistingConfigureAwaitTrue()
+        {
+
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+            var ad = new AsyncDisposable();
+#pragma warning disable AZC0013
```

Comment:
Hm, looks like we are verifying that `pragma warning disable` works in this test or am I missing something/

Reply:
The intent was to verify that `#pragma warning disable AZC0013` is enough and `AZC0012` isn't trying to handle this case. But that can be done in one test for all possible cases.